static/js/util/profile_edit.js

```diff
@@ -12,72 +15,147 @@ import moment from 'moment';
 
 // bind to this.boundTextField in the constructor of a form component
 // to update text fields when editing the profile
-export function boundTextField(key, label, error) {
-  const { updateProfile, profile } = this.props;
+// we pass in a keyset looking like this:
+//
+// ["top-level-key", index, "nested_object_key"] or just ["top_level_key"]
+export function boundTextField(keySet, label) {
```

Comment:
It would be useful to have unit tests for these functions. Would it be helpful if I worked on that?